CMakeLists.txt

```diff
@@ -66,161 +68,175 @@ check_symbol_exists(getopt_long getopt.h HAVE_GETOPT)
 
 if(WITH_CUDA)
-    # BEGIN OF WORKAROUND - enable cuda dynamic linking.
-    # Starting with Cmake 3.17 we can use
-    # set(CMAKE_CUDA_RUNTIME_LIBRARY_DEFAULT "SHARED") instead
-    set( CMAKE_CUDA_FLAGS "" CACHE STRING "" )
-    if ( CMAKE_CUDA_FLAGS )
-            list(REMOVE_ITEM CMAKE_CUDA_FLAGS "-cudart static")
-    endif()
-    string(APPEND CMAKE_CUDA_FLAGS "-cudart shared")
-    # END OF WORKAROUND
+	# BEGIN OF WORKAROUND - enable cuda dynamic linking.
+	# Starting with Cmake 3.17 we can use
+	# set(CMAKE_CUDA_RUNTIME_LIBRARY_DEFAULT "SHARED") instead
+	set(CMAKE_CUDA_FLAGS "" CACHE STRING "")
+	if(CMAKE_CUDA_FLAGS)
+		list(REMOVE_ITEM CMAKE_CUDA_FLAGS "-cudart static")
+	endif()
+	string(APPEND CMAKE_CUDA_FLAGS "-cudart shared")
+	# END OF WORKAROUND
 
 	enable_language(CUDA)
 
-    # BEGIN OF WORKAROUND - enable cuda dynamic linking.
-    if ( CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES )
-        list( REMOVE_ITEM CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES "cudart_static" )
-        list( REMOVE_ITEM CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES "cudadevrt" )
-        list( APPEND CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES "cudart" )
-    endif()
-    if ( CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES )
-        list( REMOVE_ITEM CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES "cudart_static" )
-        list( REMOVE_ITEM CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES "cudadevrt" )
-        list( APPEND CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES "cudart" )
-    endif()
-    if ( CUDA_LIBRARIES )
-        list ( REMOVE_ITEM CUDA_LIBRARIES "${CUDA_cudart_static_LIBRARY}")
-        list ( APPEND CUDA_LIBRARIES "${CUDA_CUDART_LIBRARY}")
-    endif()
-    # END OF WORKAROUND
+	# BEGIN OF WORKAROUND - enable cuda dynamic linking.
+	if(CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES)
+		list(REMOVE_ITEM CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES "cudart_static")
+		list(REMOVE_ITEM CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES "cudadevrt")
+		list(APPEND CMAKE_CUDA_HOST_IMPLICIT_LINK_LIBRARIES "cudart")
+	endif()
+	if(CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES)
+		list(REMOVE_ITEM CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES "cudart_static")
+		list(REMOVE_ITEM CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES "cudadevrt")
+		list(APPEND CMAKE_CUDA_IMPLICIT_LINK_LIBRARIES "cudart")
+	endif()
+	if(CUDA_LIBRARIES)
+		list(REMOVE_ITEM CUDA_LIBRARIES "${CUDA_cudart_static_LIBRARY}")
+		list(APPEND CUDA_LIBRARIES "${CUDA_CUDART_LIBRARY}")
+	endif()
+	# END OF WORKAROUND
 endif()
 
-add_subdirectory(dpsim)
 add_subdirectory(dpsim-models)
-
-if (WITH_VILLAS)
-	add_subdirectory(dpsim-villas)
-endif(WITH_VILLAS)
+add_subdirectory(dpsim)
 add_subdirectory(docs)
 
-include(cmake/GetGridData.cmake)
+if(WITH_VILLAS)
+	add_subdirectory(dpsim-villas)
+endif()
 
-message(VERBOSE "Compiler Release Flags: " 	${CMAKE_CXX_FLAGS_RELEASE})
-message(VERBOSE "Compiler Debug Flags: " 	${CMAKE_CXX_FLAGS_DEBUG})
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16")
```

Comment:
why is this necessary? Was this wrong before?

Reply:
The `VERBOSE` message level has been introduced by CMake 3.16 and hence is not supported with earlier CMake versions.